dj/sql/parsing/frontends/sql_string.py

```diff
@@ -0,0 +1,138 @@
+"""
+Transform a DJ AST into a sql string
+"""
+
+from functools import singledispatch
+from typing import Any
+
+from dj.sql.parsing.ast import (
+    Alias,
+    Between,
+    BinaryOp,
+    Case,
+    Column,
+    From,
+    Function,
+    Join,
+    Node,
+    Query,
+    Select,
+    String,
+    Table,
+    UnaryOp,
+    Value,
+    Wildcard,
+)
+
+
+@singledispatch
+def sql(node: Any) -> str:
+    """
+    return the ansi sql representing the sub-ast
+    """
+    raise Exception("Can only convert specific Node types to a sql string")
+
+
+@sql.register
+def _(node: UnaryOp) -> str:
+    return f"{node.op.value} {sql(node.expr)}"
```

Comment:
I think in your first approach you had `sql` implemented as a method of each node, instead of dispatcher... any reason for the change? We could implement it as `__str__` method of each node, which would simplify a lot of the logic here.

Reply:
First, it was just `sql` as you pointed out which mimics sqlglot behavior. Then, with the advent of that frontend/backend separation the reasoning was `something -> ast` = backend and `ast -> something` = frontend and `ast -> string` felt like it fit the frontend pattern. Then, using dispatch just made sense to me as a clean way to implement it (I probably would have done the same thing if it were possible to match on typedict for the sqloxide backend which would have been super clean imo). Does all of that make sense? I'm perfectly happy to have it as `__str__` instead like you said that would clean up a lot

Reply:
@betodealmeida would the `__str__` representation of a node be more than just the SQL string fragment? I'd expect it would also contain some of the other attributes of the node (maybe).

Imo, even though it's nice to be able to just look at a concrete node implementation to see how it's casted to a string, it's also nice to look in one place to see all of the casting logic. That also makes it easy to create another "frontend"--you can just copy sql_string.py and edit from there.

Reply:
+1 on @samredai point. Perhaps the only thing I am not clear on is that `ast -> string` does not really clarify what translation we are using. If it was `ast -> ansi sql` and then self.__str__() that would be clear. Maybe we should just call the `sql_string` frontend as `ansi_sql` to clarify that this is really ANSI SQL converted to string.

Reply:
This does not have to happen in the same PR. I am ok with merging in and renaming things later, if all agree.